src/operation/overlay/snap/GeometrySnapper.cpp

```diff
@@ -59,8 +59,9 @@ class SnapTransformer: public geos::geom::util::GeometryTransformer {
         using std::unique_ptr;
 
         assert(srcPts);
-        assert(srcPts->toVector());
-        LineStringSnapper snapper(*(srcPts->toVector()), snapTol);
+        std::vector<Coordinate> coords;
+        srcPts->toVector(coords);
+        LineStringSnapper snapper(coords, snapTol);
```

Comment:
I think, grand objective would be to get this bit read as:

```cpp
std::vector<Coordinate> coords = srcPts->toVector();
LineStringSnapper snapper(std::move(coords), snapTol);
```

Reply:
`LineStringSnapper` is taking `vector<T> const&`. But I see the merits of going to 

```
auto coords = srcPts->toVector();
LineStringSnapper snapper(coords, snapTol);
```

Reply:
Ironically the LineStringSnapper is not mutating the input coords anyway, and so this is just making an extra copy.  This is an example of where JTS is not fully utilzing the CoordinateSequence interface (i.e. creating an extra copy if the CS is not actually an array).